var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,166 @@
+                    'CMake/FindLibxml2QMC.cmake')
+
+    def cmake_args(self):
+        args = []
+
+        if '+mpi' in self.spec:
+            mpi = self.spec['mpi']
+            args.append('-DCMAKE_C_COMPILER={0}'.format(mpi.mpicc))
+            args.append('-DCMAKE_CXX_COMPILER={0}'.format(mpi.mpicxx))
+            args.append('-DMPI_BASE_DIR:PATH={0}'.format(mpi.prefix))
+
+        # Currently FFTW_HOME and LIBXML2_HOME are used by CMake.
+        # Any CMake warnings about other variables are benign.
+        xml2_prefix = self.spec['libxml2'].prefix
+        args.append('-DLIBXML2_HOME={0}'.format(xml2_prefix))
+        args.append('-DLibxml2_INCLUDE_DIRS={0}'.format(xml2_prefix.include))
+        args.append('-DLibxml2_LIBRARY_DIRS={0}'.format(xml2_prefix.lib))
+
+        fftw_prefix = self.spec['fftw'].prefix
+        args.append('-DFFTW_HOME={0}'.format(fftw_prefix))
+        args.append('-DFFTW_INCLUDE_DIRS={0}'.format(fftw_prefix.include))
+        args.append('-DFFTW_LIBRARY_DIRS={0}'.format(fftw_prefix.lib))
+
+        args.append('-DBOOST_ROOT={0}'.format(self.spec['boost'].prefix))
+        args.append('-DHDF5_ROOT={0}'.format(self.spec['hdf5'].prefix))
+
+        # Default is MPI, serial version is convenient for cases, e.g. laptops
+        if '~mpi' in self.spec:
+            args.append('-DQMC_MPI=0')
+
+        # When '-DQMC_CUDA=1', CMake automatically sets:
+        # '-DQMC_MIXED_PRECISION=1'
+        #
+        # Note: there is a double-precision CUDA path, but it is deprecated
+        if '+cuda' in self.spec:
+            args.append('-DQMC_CUDA=1')
+
+        # this is for the experimental mixed-prescision CPU code
+        if '+mixed' in self.spec:
+            args.append('-DQMC_MIXED_PRECISION=1')
+
+        return args
+
+    def install(self, spec, prefix):
+        """Make the install targets"""
+
+        # QMCPACK 'make install' does nothing, which causes
+        # Spack to throw an error.
+        #
+        # This install method creates the top level directory
+        # and copies the bin subdirectory into the appropriate
+        # location. We do not copy include or lib at this time due
+        # to technical difficulties in qmcpack itself.
+
+        mkdirp(prefix)
+
+        # We assume cwd is self.stage.source_path
+
+        # install manual
+        install_tree('manual', prefix.manual)
+
+        # install nexus
+        install_tree('nexus', prefix.nexus)
+
+        with working_dir(self.build_directory):
+            mkdirp(prefix)
+
+            # install binaries
+            install_tree('bin', prefix.bin)
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check_build(self):
+        """Run ctest after building binary.
+        Use 'spack install --run-tests qmcpack'.
+        It can take 24 hours to run all the regression tests.
+        We only run the unit tests and short tests."""
+        ctest('-L unit')
+        ctest('-R short')
```

Comment:
These probably need to look like:
```python
ctest('-L', 'unit')
ctest('-R', 'short')
```
Otherwise, it is essentially the same as running:
```
$ ctest "-L unit"
```
on the command line. There may be problems parsing this as a single string.